test/exchange-connector/exchangeTools.test.ts

```diff
@@ -184,3 +186,64 @@ describe("Test groupByCategory method", () => {
     });
   });
 });
+
+describe("removeRamlLinks", () => {
+  const REST_APIS: RestApi[] = [
+    {
+      id: "8888888/test-api/1.0.0",
+      name: "Test API",
+      groupId: "8888888",
+      assetId: "test-api",
+      fatRaml: {
+        classifier: "rest-api",
+        sha1: "sha1",
+        md5: "md5",
+        externalLink: "https://somewhere/fatraml.zip",
+        packaging: "zip",
+        createdDate: "today",
+        mainFile: "api.raml"
+      }
+    },
+    {
+      id: "8888888/test-api2/1.0.0",
+      name: "Test API",
+      groupId: "8888888",
+      assetId: "test-api",
+      fatRaml: {
+        classifier: "rest-api",
+        sha1: "sha1",
+        md5: "md5",
+        externalLink: "https://somewhere/fatraml2.zip",
+        packaging: "zip",
+        createdDate: "today",
+        mainFile: "api.raml"
+      }
+    }
+  ];
+
+  it("should remove fat RAML external links for all the apis", () => {
+    const apis = _.cloneDeep(REST_APIS);
+    expect(removeRamlLinks(apis)).to.not.contain("externalLink");
+  });
+
+  it("should not fail if the RAML does not have externalLink fields", () => {
+    const apis = _.cloneDeep(REST_APIS);
+    apis.forEach(apiEntry => {
+      delete apiEntry.fatRaml.externalLink;
+    });
+
+    expect(removeRamlLinks(apis)).to.not.contain("externalLink");
+  });
+
+  it("should not do anything if an empty array is passed", () => {
+    expect(removeRamlLinks([])).to.deep.equal([]);
+  });
+
+  it("should return null if null is passed", () => {
```

Comment:
Is this the desired behavior? Based on the function's usage, I would consider undefined/null to be unexpected input, and unexpected input should generally throw an error.

Reply:
that's a good point. For this story I have been not throwing errors because we don't want to halt the entire process in most cases but in this case it doesn't make sense to go forward if the input is null/undefined. I'll make the change.